spec/core/instances_spec.rb

```diff
@@ -469,10 +530,26 @@
       [[:sunny, 70, 80, :TRUE, :yes], [:overcast, 80, 85, :FALSE, :yes]]
     end
 
-    it 'adds the data to the Instances object' do
-      expect { subject.add_instances(data) }
-        .to change { subject.instances_count }
-        .by(data.count)
+    context 'when each instance is stored as an array of attribute values' do
+      it 'adds the data to the Instances object' do
+        expect { subject.add_instances(data) }
+          .to change { subject.instances_count }
+          .by(data.count)
+      end
+    end
+
+    context 'when each instance is stored as a hash of attribute values' do
+      let(:hash_data) do
+        data.map do |attribute_values|
+          subject.send(:attribute_values_to_hash, attribute_values)
```

Comment:
I'm not sure, whether it is a good idea to depend on a private method here. If for some reason it is renamed/removed, this test will break, even if the result is totally fine. Maybe it is better to just redefine the hashes explicitely?